### PR TITLE
Order History Page Search

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Added feature to search by order number on the order history page.

## Related Issue
Closes [PWA-888](https://jira.corp.magento.com/browse/PWA-888)

## Acceptance
Should be able to search by order number on the order history page.

### Verification Stakeholders
@schensley 
@dpatil-magento 

### Verification Steps
1. Go to the deployed PR's order history page.
2. Enter a valid order number in the search field. You should be able to see the order details of the given order number.
3. Enter an invalid number and you should see a message stating there are no orders yet.

## Screenshots / Screen Captures (if appropriate)

image

image

image

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/OrderHistoryPage/index.js [hunk 1]
```diff
@@ -1 +1,9 @@
-export { default } from './orderHistoryPage';
+import React from 'react';
+import { Form } from 'informed';
+import OrderHistoryPage from './orderHistoryPage';
+
+export default props => (
```
Comment (revanth0212) [inline packages/venia-ui/lib/components/OrderHistoryPage/index.js:5]: I wish we had a `wrapWithForm` HOC for such cases. A simple HOC that just wraps the given component inside an `informed form`.

**Comment by schensley** [comment]
@revanth0212 this looks good. thanks for working with me on the details. UX approved.

File: packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js [hunk 2]
```diff
@@ -30,31 +64,96 @@ const OrderHistoryPage = props => {
...
+    if (isLoadingWithoutData) {
+        return fullPageLoadingIndicator;
+    }
+
     return (
         <OrderHistoryContextProvider>
             <div className={classes.root}>
                 <Title>{title}</Title>
                 <h1 className={classes.heading}>{PAGE_TITLE}</h1>
+                <div className={classes.search}>
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:139]: Something like this:
```suggestion
                <Form onSubmit={handleSubmit} className={classes.search}>
```

File: packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js [hunk 3]
```diff
@@ -30,31 +64,96 @@ const OrderHistoryPage = props => {
...
+                    <TextInput
+                        after={resetButton}
+                        before={searchIcon}
+                        field="search"
+                        id={classes.search}
+                        placeholder={SEARCH_PLACE_HOLDER}
+                        onKeyPress={handleKeyPress}
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:146]:
```suggestion
```

File: packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js [hunk 4]
```diff
@@ -30,31 +64,96 @@ const OrderHistoryPage = props => {
...
+                    />
+                    <Button
+                        disabled={isBackgroundLoading || isLoadingWithoutData}
+                        priority={'high'}
+                        onClick={getOrderDetails}
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:151]:
```suggestion
```

**Review by tjwiebell — CHANGES_REQUESTED** [review]
Clarified what we discussed in Slack a bit more, I think this can be simplified a bit further. Ping me if you have questions 🙂

File: packages/peregrine/lib/talons/OrderHistoryPage/useOrderHistoryPage.js [hunk 5]
```diff
@@ -19,21 +24,75 @@ export const useOrderHistoryPage = (props = {}) => {
...
+    const formApi = useFormApi();
+
+    const searchText = useMemo(() => {
+        return formState.values.search;
+    }, [formState]);
+
+    const [
+        getOrdersData,
+        { loading: ordersLoading, error: getOrdersError }
+    ] = useLazyQuery(getCustomerOrdersQuery, {
```
[the remark this answers is not in the document]
Comment (revanth0212) [reply packages/peregrine/lib/talons/OrderHistoryPage/useOrderHistoryPage.js]: Right.. good suggestion @tjwiebell. I have pushed the changes to use a single query with fuzzy search. cc1b52c

File: packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js (hunk 2 above)
Comment (revanth0212) [reply packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:139]: Nice idea @tjwiebell, Unfortunately, the problem with this solution is the talon won't have access to the `searchText`. The talon needs `searchText` and so does the component. We do some if-else checking in the component that needs the `searchText` from the talon.

Comment (tjwiebell) [reply packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:139]: I've encountered the chicken/egg issue with form context and using those hooks from the parent talon, so understand what you mean. I'll try my suggestion in scope of the pagination ticket, and won't hold this one up.

**Review by tjwiebell — DISMISSED** [review]
Looks good, nice work 👍  would prefer to get rid of those event handlers if possible, but understand the limitation, so won't hold this up.

**Comment by dpatil-magento** [comment]
@revanth0212 Please check following - 
1. When there are no (zero) orders, search field is still displaying > Dont think we need it there. Not a blocker, check with UX if update needed.

image


2. When I clear order number using keyboard then there is no way for user to load all orders. Actually all order should load if user clear the field and does not start typing new search string.

<a href="https://gyazo.com/30e6692b9128e54888965b26b026e63c">Image from Gyazo</a>

**Comment by revanth0212** [comment]
> @revanth0212 Please check following -
> 
> 1. When there are no (zero) orders, search field is still displaying > Dont think we need it there. Not a blocker, check with UX if update needed.
> 
> image
> 
> 1. When I clear order number using keyboard then there is no way for user to load all orders. Actually all order should load if user clear the field and does not start typing new search string.
> 
> [Image from Gyazo](https://gyazo.com/30e6692b9128e54888965b26b026e63c)

Very good point @dpatil-magento. In my opinion, it makes sense to leave the search field in place even if the orders list is empty. @schensley what do you think is the right approach in this case.

Regarding 2, I have handled it in the latest commits. f229636 0d71cc4

**Comment by dpatil-magento** [comment]
QA Approved.

**Comment by schensley** [comment]
@revanth0212 @dpatil-magento I think it's good to leave the search box even when there are no orders. It would be ideal to have a message along the lines of "You have no orders" (or the like) where the orders would/will appear.

**Review by tjwiebell — APPROVED** [review]

**Pull request merged**